app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -101,6 +110,45 @@ class MessageActivity : AppCompatActivity() {
             }
         }
 
+        save.setOnClickListener {
+            try {
+                val inputManager: InputMethodManager = this.getSystemService(Context.INPUT_METHOD_SERVICE)
+                        as InputMethodManager
+                inputManager.hideSoftInputFromWindow(content.windowToken, InputMethodManager.SHOW_FORCED)
+
+                val sdf = SimpleDateFormat("dd-MMM-yyyy-hh_mm_aaa")
+                val fileName = "badge-magic-" + sdf.format(Calendar.getInstance().time).toString()
+                val os = FileOutputStream(fileName, true)
+                var text = "null"
+                var drawableID = "null"
+                var flashSelected = false
+                var marqueeSelected = false
```

Comment:
There is no need of flashSelected and marqueeSelected

Reply:
Actually I'm using POJO and then will be passing all them at once in the moshiConverter

Reply:
But these are not needed to represent the item

Reply:
This is just so that incase that function is not enabled it goes to the default declared value.

Reply:
flash.isChecked and marquee.isChecked have the boolean already. You don't need to take new variables for it I guess

Reply:
I'd suggest you to store the hexadecimal strings of the current configuration. You are storing the text and drawable in String format which are not readable by the LED Badge or the Preview in the app. Store the hexadecimal Strings which can be read by both

Reply:
Yeah I'll do that... I think it will be good for future uses also.